lib/ui/window/platform_configuration.cc

```diff
@@ -77,12 +86,43 @@ void PlatformConfiguration::DidCreateIsolate() {
   report_timings_.Set(tonic::DartState::Current(),
                       Dart_GetField(library, tonic::ToDart("_reportTimings")));
 
-  // TODO(loicsharma): This should only be created if the embedder enables the
-  // implicit view.
-  // See: https://github.com/flutter/flutter/issues/120306
-  windows_.emplace(kImplicitViewId,
-                   std::make_unique<Window>(
-                       kImplicitViewId, ViewportMetrics{1.0, 0.0, 0.0, -1, 0}));
+  library_.Set(tonic::DartState::Current(),
+               Dart_LookupLibrary(tonic::ToDart("dart:ui")));
+}
+
+void PlatformConfiguration::AddWindowRecord(int64_t view_id) {
+  windows_.emplace(
+      view_id, std::make_unique<Window>(library_, view_id,
+                                        ViewportMetrics{1.0, 0.0, 0.0, -1, 0}));
+}
+
+void PlatformConfiguration::AddView(int64_t view_id) {
+  FML_DCHECK(view_id != kFlutterImplicitViewId);
+  AddWindowRecord(view_id);
```

Comment:
Should we check that `dart_state` is valid before calling `AddWindowRecord`? Same comment for `RemoveView`'s logic

Reply:
Don't we check `dart_state` with the if clause? Do you mean if the dart state is invalid we should not silently return but log an error?

Reply:
My concern is that `PlatformConfiguration`'s state is updated (by `AddWindowRecord`) even if `dart_state` is invalid. That could mean that `PlatformConfiguration`'s state no longer matches `dart_state`. I *think* this is ok, but perhaps it'd be better to do the check first anyways?

Reply:
To reconfirm, the question is why we allow modifying `windows_` when `dart_state` is invalid. The platform configuration states and the platform dispatcher states should be kept in sync. An alternative solution is to `FML_DCHECK` the validity of `dart_state`.

I've thought about it too, but I don't think it's that concerning because:
* Existing methods have been behaving in the same way and running fine.
* If I understand correctly, `PlatformConfiguration` is supposed to exist only when the Dart state exists. The Dart state might become invalid when then app starts shutting down, and in that case we'd prefer robustness over data correctness (since the app's ending).